cnf-certification-test/lifecycle/suite.go

```diff
@@ -293,3 +302,37 @@ func testHighAvailability(env *provider.TestEnvironment) {
 		gomega.Expect(0).To(gomega.Equal(len(badStatefulSet)))
 	})
 }
+
+// testPodsRecreation tests that pods belonging to deployments and statefulsets are re-created and ready in case a node is lost
+func testPodsRecreation(env *provider.TestEnvironment) {
+	ginkgo.By("Testing node draining effect of deployment")
+	ginkgo.By("Testing initial state for deployments")
+	claimsLog := podsets.WaitForAllPodSetReady(env, timeoutPodSetReady)
+	tnf.ClaimFilePrintf("%s", claimsLog)
+	for n := range podsets.GetAllNodesForAllPodSets(env) {
+		defer podrecreation.CordonCleanup(n) //nolint:gocritic // The defer in loop is intentional, calling the cleanup function once per node
```

Comment:
This won't actually fire until the scope of `testPodsRecreation` is completed.  Defer in a loop exits whenever func scope is complete.

Reply:
Correct, this is the expected scenario. When the function exits normally or abnormally, the uncordon is executed to put the cluster back into a good state for each node that was cordoned. There is another uncordon in the normal flow. In case everything goes well, this defer will uncordon an already uncordoned node with no effects.

Reply:
Okay sounds good.  I'm thinking this works for now and if there's any problems for the defer maybe we can address later.  I'm fine with it for now.  I just know that defer in a loop has bitten me before so I'm cautious.  😄